modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java

```diff
@@ -404,13 +405,38 @@ public CompletableFuture<JobExecution<ComputeJobDataHolder>> submitColocatedInte
             List<DeploymentUnit> units,
             String jobClassName,
             JobExecutionOptions options,
-            @Nullable CancellationToken cancellationToken,
-            @Nullable ComputeJobDataHolder arg) {
+            @Nullable ComputeJobDataHolder arg,
+            @Nullable CancellationToken cancellationToken
+    ) {
         return primaryReplicaForPartitionByTupleKey(table, key)
                 .thenApply(primaryNode -> executeOnOneNodeWithFailover(
                         primaryNode,
                         new NextColocatedWorkerSelector<>(placementDriver, topologyService, clock, table, key),
-                        units, jobClassName, options, cancellationToken, arg
+                        units, jobClassName, options, arg, cancellationToken
+                ));
+    }
+
+    @Override
+    public CompletableFuture<JobExecution<ComputeJobDataHolder>> submitPartitionedInternal(
+            TableViewInternal table,
+            int partitionId,
+            List<DeploymentUnit> units,
+            String jobClassName,
+            JobExecutionOptions jobExecutionOptions,
+            @Nullable ComputeJobDataHolder arg,
+            @Nullable CancellationToken cancellationToken
+    ) {
+        ExecutionOptions options = ExecutionOptions.builder()
+                .priority(jobExecutionOptions.priority())
+                .maxRetries(jobExecutionOptions.maxRetries())
+                .partition(new HashPartition(partitionId))
+                .build();
```

Comment:
```suggestion
                .build();

```

Separate multi-line statements for readability.